airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresUtils.java

```diff
@@ -5,11 +5,17 @@
 package io.airbyte.integrations.source.postgres;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import java.time.Duration;
 
 public class PostgresUtils {
 
   private static final String PGOUTPUT_PLUGIN = "pgoutput";
 
+  // These values cannot be too short. In local unit tests, Debezium is fast.
+  // But in reality, the initial snapshot takes longer to run.
+  public static final Duration CDC_FIRST_RECORD_TIMEOUT = Duration.ofSeconds(30L);
```

Comment:
Hi @tuliren - I'm very excited for this change! Is it possible to make it configurable though? We have a postgres source for a large database that receives very frequent updates to large records. We seem to be hitting the existing 5 minute timeout currently. I plan to look into it more next week, but perhaps others would find value to being able to tweak the timeout?

Reply:
Thank you for letting us know.

> We seem to be hitting the existing 5 minute timeout currently

You mean after waiting for 5 minutes, the connector cannot get any new data from the database even though there should be lots of updates, right?

The current approach in this PR does not work. But I think we have a solution for your case.

The timeout is needed for a few reasons:
1. It can take sometime for Debezium to connect to the binlog. For Postgres, however, this is usually very fast, based on my local test and test with AWS RDS. So I think it is not a concern for Postgres. For other databases like MySQL, it can be much longer. The amount of time needed here is likely a constant depending on the database.
2. Once Debezium starts to process the write-ahead logs, it may need to filter out some logs that have already been processed in a previous run. The number of logs to filter out is unknown, and the time Debezium needs can vary a lot. For example, when there are 1 million events to filter out, it can take 1-2 minutes. When Debezium is in this stage, the connector should keep waiting. We think we have a solution to automatically detect this. So we don't need to configure the timeout for this reason.
3. When the connector is set up for the first time, Debezium needs to snapshot the existing databases. I think we can detect this as well, because the connection has an empty state when it is first set up. So no timeout parameter is needed for this either.

If our implementation works, you probably don't need to configure the timeout for your use case. The only reason we may need to expose this parameter is for bullet point 1.

In terms of making this timeout configurable, our thinkings are:
1. If we can find a good way to automatically deal with all the edge cases, we will just do that and won't expose it as a configurable parameter. The rationale is that low level parameters like this can confuse users, or even backfire by disrupting the connection if a wrong or bad value is set.
2. If we cannot resolve all edge cases, we will expose it as a parameter and add documentation for it.

So if our implementation turns out to not work, we will do that.

I should have some updates on this PR today or early next week.